### PR TITLE
fix tests in test262/built-ins/String/fromCodePoint/*

### PR DESCRIPTION
**Pull request by ian4hu**
Additional tests for #731

### CODE REVIEW

File: testsrc/test262.properties [hunk 1]
```diff
@@ -848,15 +848,15 @@ built-ins/String
...
-    ! fromCodePoint/argument-is-not-integer.js
-    ! fromCodePoint/argument-not-coercible.js
-    ! fromCodePoint/arguments-is-empty.js
-    ! fromCodePoint/fromCodePoint.js
-    ! fromCodePoint/length.js
-    ! fromCodePoint/name.js
-    ! fromCodePoint/number-is-out-of-range.js
-    ! fromCodePoint/return-string-value.js
-    ! fromCodePoint/to-number-conversions.js
+    # fromCodePoint/argument-is-not-integer.js
```
Comment (rbri) [inline testsrc/test262.properties:851]: please remove the lines complety

File: testsrc/test262.properties [hunk 2]
```diff
@@ -848,15 +848,15 @@ built-ins/String
     ! prototype/toLowerCase/special_casing_conditional.js
     ! prototype/Symbol.iterator/this-val-non-obj-coercible.js
     # not implemented
```
Comment (rbri) [inline testsrc/test262.properties:850]: you can remove this line also

File: src/org/mozilla/javascript/NativeString.java [hunk 3]
```diff
@@ -242,7 +242,13 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
                     }
                     int[] codePoints = new int[n];
                     for (int i = 0; i != n; i++) {
-                        codePoints[i] = ScriptRuntime.toInt32(args[i]);
+                        Object arg = args[i];
+                        int codePoint = ScriptRuntime.toInt32(arg);
+                        double num = ScriptRuntime.toNumber(arg);
+                        if (!ScriptRuntime.eqNumber(num, codePoint) || !Character.isValidCodePoint(codePoint)) {
+                            throw rangeError("invalid code point " + ScriptRuntime.toString(arg));
```
Comment (rbri) [inline src/org/mozilla/javascript/NativeString.java:249]: I think the error messages start usually with an uppercase letter

**Comment by gbrail** [comment]
Thanks! Can you address those two small suggestions above please?

I spent a bunch of time refreshing my memory on how Infinity and NaN behave in Java but it looks like the test cases have it covered.

Also (not for this PR) if someone wants something to do, we could probably use a "ScriptRuntime.toLegalInt32" method that converts an Object into an int32 but throws the appropriate exception if the integer is not valid. It would probably do to make this code slightly more efficient and I bet we could clean up 10 other places as well.

File: src/org/mozilla/javascript/NativeString.java (hunk 3 above)
Comment (ian4hu) [reply src/org/mozilla/javascript/NativeString.java:249]: u r right.

**Comment by ian4hu** [comment]
> Thanks! Can you address those two small suggestions above please?
> 
> I spent a bunch of time refreshing my memory on how Infinity and NaN behave in Java but it looks like the test cases have it covered.
> 
> Also (not for this PR) if someone wants something to do, we could probably use a "ScriptRuntime.toLegalInt32" method that converts an Object into an int32 but throws the appropriate exception if the integer is not valid. It would probably do to make this code slightly more efficient and I bet we could clean up 10 other places as well.

You are not the only one who spent time digging into the `ScriptRuntime.toXXX`

**Comment by gbrail** [comment]
Thanks for contributing this!

**Pull request merged**